app/gameengine.py

```diff
@@ -3385,10 +3436,13 @@ def archive_hand_continuation():
                         "continuation": self.continue_resolving_effects,
                     })
             case EffectType.EffectType_Draw:
-                amount = effect["amount"]
+                amount = effect.get("amount", len(effect_player.hand))
                 if str(amount) == "last_card_count":
                     amount = self.last_card_count
                     self.last_card_count = 0
+                max_in_hand = effect.get("max_in_hand")
+                if max_in_hand:
```

Comment:
if max_in_hand won't quite work if we ever want to support 0 as a max_in_hand option.

Reply:
Actually this comment doesn't make sense, and I see where I got confused.
Max in hand, to me, sounds more like a condition, whereas you're using this really as a "draw to hand size"
Maybe we should rename this.